pkg/joinserver/joinserver.go

```diff
@@ -320,10 +320,21 @@ func (js *JoinServer) HandleJoin(ctx context.Context, req *ttnpb.JoinRequest) (r
 		},
 		func(ctx context.Context, dev *ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error) {
 			if dn, ok := auth.X509DNFromContext(ctx); ok {
-				if dev.NetID == nil {
+				netID := dev.NetID
+				if netID == nil {
+					appSettings, err := js.applicationActivationSettings.GetByID(ctx, dev.ApplicationIdentifiers, []string{
```

Comment:
it would be great if only 1 call to the registry would be required, because currently the call on L515 can be executed as well, even though we already have enough data